prow/cmd/deck/main.go

```diff
@@ -345,17 +345,15 @@ func (c *filteringProwJobLister) ListProwJobs(selector string) ([]prowapi.ProwJo
 
 	var filtered []prowapi.ProwJob
 	for _, item := range prowJobList.Items {
-		if item.Spec.Refs == nil && len(item.Spec.ExtraRefs) == 0 {
-			// periodic jobs with no refs cannot be filtered
-			filtered = append(filtered, item)
-			continue
-		}
-
 		refs := item.Spec.Refs
 		if refs == nil {
-			refs = &item.Spec.ExtraRefs[0]
+			if len(item.Spec.ExtraRefs) > 0 {
+				refs = &item.Spec.ExtraRefs[0]
+			} else {
```

Comment:
We don't need this `else`, do we?

Reply:
Do do, else we'll may get a NPE in L356

Reply:
I don't see how that helps. What happens for a periodic with no `item.Spec.Refs` or `item.Spec.ExtraRefs`? The `refs` var remains `nil`.

Reply:
The `else` branch is for the `if len(item.Spec.ExtraRefs) > 0 ` condition, not the `refs == nil`.

If this wouldn't work the test would blow up, the prowjobs there dont have any refs.

Reply:
Right, that's how I am reading it. So if `item.Spec.Refs` is `nil` and `item.Spec.ExtraRefs` is `nil`, then from the assignment `refs := item.Spec.Refs`. the `refs` is `nil` and will remain `nil`?

Reply:
No, it gets set to something non-nil on line 353